clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -440,6 +440,7 @@ private void handleCompletedReceives(List<ClientResponse> responses, long now) {
             correlate(req.request().header(), header);
             if (!metadataUpdater.maybeHandleCompletedReceive(req, now, body))
                 responses.add(new ClientResponse(req, now, false, body));
+            inFlightRequests.completeNext(source);
```

Comment:
@junrao Maybe we can catch all exceptions here and wrap it using ```ClientResponse``` with a null body and let the upper layer handle it?